### DISCUSSION TITLE
Multithreaded option?

### DISCUSSION THREAD

**Discussion by andrew-bydlon** [tjko/jpegoptim#99] [General]
Is there no option to multithread the procedure.  For batch processing on modern computers, this seems essential.

**Comment by andrew-bydlon** [comment]
Here is a simple way to handle it:

N = 16 # Number of threads
for file in * ; do #where * is the directory of interest)
   ((i=i%N)); ((i++==0)) && wait
   jpegoptim $file & 
done

**Comment by TheLastGimbus** [comment]
This is really needed, seems dumb simple to implement, and would increase speed by number of cores

Currently, as `jpegoptim` takes just a list of files to compress, you could write a script to stupidly cut the list of them to N number of cores and launch as many `jpegoptim` processes

**Comment by tjko** [comment]
Initial parallel processing support has been added (to 1.5.0beta). It can be enabled using _--workers=maxthreads_ option.

It has not been tested much, so any feedback/test results would be much appreciated...
